apps/ftu/js/wifi.js

```diff
@@ -230,8 +231,10 @@ var WifiUI = {
     // Retrieve SSID from dataset
     var ssid = event.target.dataset.ssid;
 
+    var selectedNetwork = WifiManager.getNetwork(ssid);
```

Comment:
getNetwork can be async, as it now calls getNetworks(callback) which will call scan if we've not yet populated this.networks. In that case we can expect undefined (if scan is in-progress). And if ssid doesnt match we'll get null back. This all seems racy and in need of some refactor. I know that starts to go beyond the scope of this bug but lets at least not introduce new race conditions, and file a follow up to promise-ize (or whatever) the networks management stuff.